[RUM] add v4 section to browser SDK upgrade guide (#12760)
DataDog/documentation: content/en/real_user_monitoring/guide/browser-sdk-upgrade.md

#### Minimal Typescript version support

Browser SDK v4 is not compatible with TypeScript earlier than v3.8.2. If you use TypeScript, ensure that the version is at least v3.8.2.

#### Tags syntax

The `version`, `env`, and `service` initialization parameters are sent as tags to Datadog. The Browser SDK slightly sanitizes them, to ensure that they don't generate multiple tags, and prints a warning if those values don't meet the tag requirements syntax.

#### Stricter initialization parameters typing

TypeScript types representing initialization parameters are stricter and may reject previously accepted unsupported parameters. If you get type-checking errors, ensure you are providing supported initialization parameters.

#### Privacy options precedence

When multiple privacy options are specified on the same element, Datadog applies the most restrictive option to avoid unexpectedly leaking sensitive data. For example, if both `dd-privacy-allow` and `dd-privacy-hidden` classes are specified on the same element, it is hidden instead of allowed.

#### Action names computation

> Remove text of children with `data-dd-action-name` attribute when computing action name from inner text.

Reviewer: ```suggestion
When computing action names, the Browser SDK removes text of child elements with the `data-dd-action-name` attribute from inner text.
```

> Ex:

Reviewer: ```suggestion
For example, for the following `container` element, where previously the computed action name would be `Container sensitive data`, in v4 the computed action name is `Container`:
```

```html
<div id="container">
  Container
  <div data-dd-action-name="sensitive">sensitive data</div>
</div>
```

> When computing the action name for the `container` element:
> - previous name: `Container sensitive data`
> - new name: `Container`

Reviewer: ```suggestion
```

### Minimal Typescript version support

The Browser SDK is now incompatible with TypeScript below v3.8.2. If you are using TypeScript, please make sure that the version you are using is greater or equal to v3.8.2.

### Tags syntax

version, env and service initialization parameters are sent as tags to Datadog. The Browser SDK is now slightly sanitizes them (to ensure that they don't unexpectedly generate multiple tags) and prints a warning if those values don't meet the tag requirements syntax.

### Stricter initialization parameters typings

TypeScript types representing initialization parameters are now stricter and may reject previously accepted unsupported parameters. If you get typechecking issues, please make sure you are only providing supported initialization parameters.

### Privacy options precedence

When multiple privacy options are specified on the same element, we now apply the most restrictive one to avoid unexpectedly leaking privacy (ex: if both `dd-privacy-allow` and `dd-privacy-hidden` classes are specified on the same element, we now consider it hidden instead of allow).

### Action names computation

> When computing automatic actions target name on elements without `data-dd-action-name` attribute, we are using the "inner text" of this element. If it contains some elements that specify `data-dd-action-name`, their text is now removed inside the ancestor inner text.

Reviewer: Struggling a bit with this one. Could we instead open with identifying the change the person might see? Something like:

```suggestion
Some automatic action names (ones without a `data-dd-action-name` attribute) might change within Datadog App because they are generated using the _inner text_ of this element. If the action contains some elements that specify `data-dd-action-name`, their text is removed inside the ancestor inner text.
```
Is the last sentence necessary? I replaced the word "it" with "the action" but I wasn't actually sure if that's correct. Do we really need to be this specific about how the action name will be different now? If so, can we be clearer (maybe provide an example)?

Author: I have reworked this part, let me know what you think

Reviewer: Now I get it. I gave it a grammar rewrite, and committed. Are you ready to publish this?

Author: All good, thanks a lot for your review!

This might slightly change automatic action names shown in the Datadog App.

## Removals

### XHR `_datadog_xhr` field

The Browser SDK previously used a `_datadog_xhr` property on `XMLHttpRequest` objects representing its internal state. This property has been removed without replacement as it wasn't intended to be used externally.

### `proxyHost` initialization parameter

> The `proxyHost` initialization parameter have been removed. Make sure to use the simpler `proxyUrl` initialization parameter instead.

Reviewer: ```suggestion
The `proxyHost` initialization parameter has been removed. Use the `proxyUrl` initialization parameter instead.
```

### Privacy options support

> `input-ignored` and `input-masked` are no longer valid privacy options and should be replaced with the `mask-user-input` privacy option. Specifically, you should replace:

Reviewer: ```suggestion
The options `input-ignored` and `input-masked` are no longer valid privacy options. Instead, use the `mask-user-input` privacy option. Specifically, replace:
```
